### PR TITLE
#157354404 UI tests

### PR DESCRIPTION
**Pull request by Rwothoromo**
#### What does this PR do?
Adds UI tests.

#### Description of Task to be completed?
- Add UI test for `MainActivty`.
- Add UI test for `GithubUserProfile`.
- Upload coverage reports to CodeClimate.

#### How should this be manually tested?
Run `./gradlew connectedCheck` or `./gradlew createDebugCoverageReport` in your terminal.
The User interface will be automatically and visibly be interacted with, on your device/emulator, by the testing utility.
Also, test coverage reports will be generated.

#### Any background context you want to provide?
There were no UI tests written for the Activities created.

#### What are the relevant pivotal tracker stories?
[#157354404](https://www.pivotaltracker.com/story/show/157354404)

#### Screenshots (if appropriate)?
The UI tests will be visible on the device/emulator screen in a manner similar to this:

ui-tests

#### Questions?
How can I fix the failing test [clickableRecyclerViewItems](https://github.com/Rwothoromo/Android-Codelab/blob/ft-157354404-ui-tests/app/src/androidTest/java/com/example/rwothoromo/developers/MainActivityTest.java#clickableRecyclerViewItems)?

### CODE REVIEW

File: app/src/androidTest/java/com/example/rwothoromo/developers/GithubUserProfileTest.java [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+    private CountingIdlingResource countingResource;
+
+    /**
+     * Register any resource that needs to be synchronized with Espresso before the test is run.
+     *
+     * @throws Exception when method fails
+     */
+    @Before
+    public void setUp() throws Exception {
```
Comment (idrisadetunmbi) [inline app/src/androidTest/java/com/example/rwothoromo/developers/GithubUserProfileTest.java]: I am not sure you need to register an idling resource for this too, it does not involve any network calls.

File: app/src/androidTest/java/com/example/rwothoromo/developers/MainActivityTest.java [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+    @Test
+    public void clickableRecyclerViewItems() throws InterruptedException {
+
+        onView(withId(R.id.recyclerView)).check(matches(isDisplayed()));
+        Thread.sleep(5000);
+
+        Intents.init();
+
+        // Scroll to an item at a position and click on it.
+        final int mockPosition = 0;
```
Comment (idrisadetunmbi) [inline app/src/androidTest/java/com/example/rwothoromo/developers/MainActivityTest.java]: Remember our discussion about this, you cannot be so sure there will always be an item at this position (or any position at all) if your tests are dependent on network calls (or more precisely, unpredictable data).

File: app/src/androidTest/java/com/example/rwothoromo/developers/GithubUserProfileTest.java (hunk 1 above)
Comment (Rwothoromo) [reply app/src/androidTest/java/com/example/rwothoromo/developers/GithubUserProfileTest.java]: Thank you @idrisadetunmbi, there is actually no need for an idling resource here, so I'll remove it.

File: app/src/androidTest/java/com/example/rwothoromo/developers/MainActivityTest.java (hunk 2 above)
Comment (Rwothoromo) [reply app/src/androidTest/java/com/example/rwothoromo/developers/MainActivityTest.java]: Oh yes, I'll try mocking this instead.

Comment (Rwothoromo) [reply app/src/androidTest/java/com/example/rwothoromo/developers/MainActivityTest.java]: I have updated the `IdlingResource` correctly, incremented it before the API and decremented it `onResponse` and `onFailure`.

All `UI tests` are now working.

**Pull request merged**